### ISSUE TITLE
Amplify api mock not working. Failed to start API Mocking. Running cleanup tasks. 🛑 Failed to start API Mocking.. Reason: Could not get resource 

### ISSUE THREAD

**Issue by owboateng** [aws-amplify/amplify-category-api#1032]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

yarn

### If applicable, what version of Node.js are you using?

v16.3.0

### Amplify CLI Version

10.5.1

### What operating system are you using?

Mac, Intel Iris Plus Graphics, MacOS Ventura

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No, new project.

### Amplify Categories

api

### Amplify Commands

Not applicable

### Describe the bug

New Amplify project.
I am able to push and pull successfully.
Everything works well in the cloud but once I try to mock the api, I get an error:

```
✅ GraphQL schema compiled successfully.
Failed to start API Mocking. Running cleanup tasks.
🛑 Failed to start API Mocking.. Reason: Could not get resource XXXXXX
Session Identifier: be90b221-254b-4813-bfc1-ad689d5d87c9
```

Resource XXXXX is a graphQL schema which exists and compiles.

### Expected behavior

That I am able to mock API

### Reproduction steps

Create new Amplify project.
Create multiple schemas in the schema directory.
Try to mock the API.
Note I have about 9 different .graphql files with different models in each.
Sometimes it works when I remove all the .graphql files and leave only one of them

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Project Identifier

29831d4612dedd8e9b22e688944f7e6a

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by josefaidt** [comment]
Hey @owboateng :wave: thanks for raising this! This appears similar to https://github.com/aws-amplify/amplify-category-api/issues/560, which has since been fixed, however we have been able to reproduce this after pushing a split schema. Taking a further look at this one

**Comment by owboateng** [comment]
> Hey @owboateng 👋 thanks for raising this! This appears similar to #560, which has since been fixed, however we have been able to reproduce this after pushing a split schema. Taking a further look at this one

Hey @josefaidt , thanks for acknowledging this. Looking forward for a fix soon.

**Comment by dwamianm** [comment]
> 

@owboateng @josefaidt  I can confirm that this issue exists in 10.5.1...rolling back to 10.3.2 fixes the issue. I tried re-ordering types as proposed in #560 but with no success since the error is not very specific.

**Comment by phani-srikar** [comment]
Hi folks, thanks for reporting this. We are tracking this issue in https://github.com/aws-amplify/amplify-category-api/issues/1014. Please follow that thread for updates.

**Comment by tmli3b3rm4n** [comment]
v10.6.1, mac m1 silicon.  The patch didn't resolve my issue.

```
✅ GraphQL schema compiled successfully.

Edit your schema at /Users/macmac/go/src/github.com/ThatLeague/battlegit/amplify/backend/api/battlegit/schema.graphql or place .graphql files in a directory at /Users/macmac/go/src/github.com/ThatLeague/battlegit/amplify/backend/api/battlegit/schema

Failed to start API Mocking. Running cleanup tasks.
🛑 Failed to start API Mocking.. Reason: Could not get resource Team


Session Identifier: caa04b26-6018-446f-a246-5e2329cd6a29

✅ Report saved: /var/folders/r1/bfypxjrn4c56t1qppzqlgg780000gn/T/battlegit/report-1673390064252.zip

✔ Done

Project Identifier: 333a9cfc635640cfaec2a93c7b0bbe98
```

**Comment by basitanwer** [comment]
Still having this issue in v10.7.0.


```bash
Failed to start API Mocking. Running cleanup tasks.
MockProcessFault: Failed to start API Mocking.. Reason: Could not get resource <XYZ>
    at APITest.start (/snapshot/repo/build/node_modules/amplify-util-mock/lib/api/api.js:151:13)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async start (/snapshot/repo/build/node_modules/amplify-util-mock/lib/api/index.js:18:5)
    at async Object.run (/snapshot/repo/build/node_modules/amplify-util-mock/lib/commands/mock/api.js:21:5)
    at async Object.executeAmplifyCommand (/snapshot/repo/build/node_modules/amplify-util-mock/lib/amplify-plugin-index.js:47:3)
    at async executePluginModuleCommand (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/execution-manager.js:142:5)
    at async executeCommand (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/execution-manager.js:40:9)
    at async Object.run (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/index.js:117:5) {
  classification: 'FAULT',
  options: {
    message: 'Failed to start API Mocking.. Reason: Could not get resource <XYZ>',
    link: 'https://docs.amplify.aws/cli/graphql/troubleshooting/'
  },
  downstreamException: undefined,
  toObject: [Function (anonymous)],
  details: undefined,
  resolution: undefined,
  code: undefined,
  link: 'https://docs.amplify.aws/cli/graphql/troubleshooting/'
}
```

**Comment by dwamianm** [comment]
@phani-srikar . I was in #1014 as well which was closed but this issue still exists and is starting to cause some major issues. Please help!!

**Comment by lewisdonovan** [comment]
I'm seeing the same issue in 10.7.2. The schema compiles fine, but mocking throws the "Could not get resource" error.

**Comment by jakubcolony** [comment]
Still experiencing that in both 10.6.1 and 10.7.2

**Comment by s-ueno** [comment]
It also occurs in version 10.7.3.
I cannot perform the test.

The following error message appears, indicating that I cannot access the schema added in version 10 or later of the amplify cli:

"Failed to start API Mocking.. Reason: Could not get resource XXXX"

**Comment by apachler** [comment]
I'm seeing the same issue in 10.7.3. The schema compiles fine, but mocking throws the "Could not get resource" error.

**Comment by jmelendev** [comment]
> I'm seeing the same issue in 10.7.3. The schema compiles fine, but mocking throws the "Could not get resource" error.

Same here. Were you able to find a fix?

**Comment by apachler** [comment]
No, sadly not.

**Comment by dwamianm** [comment]
This issue has been fixed for me in 10.8.1. 

However, there is now a new issue where GraphiQL is not showing the schema on the side. I will go look for similar issues but this one is now closed for me.

**Comment by phani-srikar** [comment]
A fix has been released for this issue in `10.8.x`. Please upgrade your version and check. Thanks.

**Comment by charlesfayal** [comment]
> A fix has been released for this issue in `10.8.x`. Please upgrade your version and check. Thanks.

Still not working, showing no schema

Seeing error in inspect, not sure if it's relevant
"main.54e56dbc.js:2 Uncaught (in promise) Error: Error when generating OIDC token: Right-hand side of 'instanceof' is not an object"

**Comment by chrisbonifacio** [comment]
@charlesfayal have you tried clearing the mock data before starting the mock server?

**Comment by charlesfayal** [comment]
Yup!On Mar 3, 2023, at 8:34 AM, Chris Bonifacio ***@***.***> wrote:﻿






*Message sent from a system outside of UConn.*




@charlesfayal
 have you tried clearing the mock data before starting the mock server?
—
Reply to this email directly, 
view it on GitHub, or 
unsubscribe.
You are receiving this because you were mentioned.Message
 ID: ***@***.***>

**Comment by jvaczcorporation** [comment]
Can anyone tell me if you need to install some specific version of python for the mock OpenSearch to work? I'm on version 3.8.16.
